lib/customisation.nix

```diff
@@ -296,4 +296,32 @@ rec {
       };
     in self;
 
+
+
+  /* Apply a function to an attrset or potential args, automatically filtering
+     the set to those the function can accept.
+     NOTE: This is strictly meant to be used with functions that accept attrsets
+     as their argument, and is best suited for use with `setFunctionArgs'.
```

Comment:
Then the name is too general.
I expected a function called `apply` to be like `merge = a: b: a // b;`, so `apply = f: a: f a`. Something you may use in folding function.

Reply:
How about :

`applyArgs` ( my favorite ) 

`fnApply`

`applyFn`

`functionApply`

Reply:
I went with `applyArgs`. Let me know if that needs further review.

Reply:
I like that it's coherent with `setFunctionArgs`. :+1: